src/pipeline/pipeline.go

```diff
@@ -102,7 +102,8 @@ func (p *pipeline) input(name string) error {
 func (p *pipeline) inject(name string) error {
 	switch {
 	case strings.HasPrefix(name, "s3://"):
-		bucket, err := s3utils.NewBucket(name)
+		bucket, err := s3utils.GetBucket(name)
+		client := s3.New(nil)
```

Comment:
It's kind of weird indeed. It's just `New` takes a pointer to a struct to override the default configuration and we don't need to override the default config.

I'll replace it with a `NewClient` function in s3utils in case we have to override the default config but other than that, it's sounds fine to me.